lib/components/joystick/Joystick_action.dart

```diff
@@ -0,0 +1,252 @@
+import 'dart:math';
+
+import 'package:flame/components/joystick/joystick_component.dart';
+import 'package:flame/components/joystick/joystick_events.dart';
+import 'package:flame/gestures.dart';
+import 'package:flame/sprite.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+enum JoystickActionAlign { TOP_LEFT, BOTTOM_LEFT, TOP_RIGHT, BOTTOM_RIGHT }
+
+class JoystickAction {
```

Comment:
add `with` gesture mixins that it uses

Reply:
The `mixin` that could be used there is the `MultiTouchDragDetector`, but it can only be applied in classes that extend Game

Reply:
Ah, yes, forgot that.